### PR TITLE
Bug 1771335: set .dockerconfigjson into containers/image auth file path instead of…

### PR DESCRIPTION
**Pull request by gabemontero**
… legacy auth  file path

/assign @adambkaplan 

@openshift/openshift-team-developer-experience @bparees @pweil- FYI

### CODE REVIEW

File: pkg/build/builder/cmd/dockercfg/cfg.go [hunk 1]
```diff
@@ -66,7 +66,8 @@ func (h *Helper) GetDockerAuthSearchPaths(authType string) []string {
 // SetSystemContextFilePath properly seeds the container/image SystemContext
 // with the authentication file based on the format implied by the file name
 func SetSystemContextFilePath(sc *types.SystemContext, path string) {
-	if filepath.Base(path) == dockerConfigFileKey {
+	base := filepath.Base(path)
+	if base == dockerConfigFileKey || base == DockerConfigJsonKey {
```
Comment (adambkaplan) [inline pkg/build/builder/cmd/dockercfg/cfg.go]: @gabemontero is `.dockercfg` considered the legacy auth file path by c/image?

Comment (gabemontero) [reply pkg/build/builder/cmd/dockercfg/cfg.go]: yes

Comment (adambkaplan) [reply pkg/build/builder/cmd/dockercfg/cfg.go]: Makes a bit more sense to flip the logic:

```
if base == DockerConfigKey {
    sc.LegacyFormatAuthFilePath = path
    return
}
sc.AuthFilePath = path
```

Comment (gabemontero) [reply pkg/build/builder/cmd/dockercfg/cfg.go]: logic flipped @adambkaplan

**Comment by gabemontero** [comment]
origin verification is https://github.com/openshift/origin/pull/24148

**Comment by gabemontero** [comment]
installer ignition flake for image eco

/test e2e-aws-image-ecosystem

**Pull request merged**